lib/components/scanner/scanner_screen.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:seeds/components/scanner/components/qr_code_view.dart';
+import 'package:seeds/components/scanner/interactor/viewmodels/scanner_bloc.dart';
+import 'package:seeds/utils/string_extension.dart';
+
+/// Scanner SCREEN
+class ScannerScreen extends StatefulWidget {
+  final ScannerBloc _scannerBloc = ScannerBloc();
+  final ValueSetter<String> onCodeScanned;
+
+  ScannerScreen({Key? key, required this.onCodeScanned}) : super(key: key);
+
+  void scan() => _scannerBloc.add(const Scan());
+
+  void showLoading() => _scannerBloc.add(const ShowLoading());
+
+  @override
+  _ScannerScreenState createState() => _ScannerScreenState();
+}
+
+class _ScannerScreenState extends State<ScannerScreen> {
+  final GlobalKey _qrKey = GlobalKey(debugLabel: 'QR');
```

Comment:
I don't think this global key be necessary too. I let a issue in the repo to know if this is a requirement.
https: //github.com /juliuscanute /qr_code_scanner /issues/ 472